spec/controllers/api/enterprise_fees_controller_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'spec_helper'
+
+module Api
+  describe EnterpriseFeesController, type: :controller do
+    include AuthenticationWorkflow
+
+    let(:unreferenced_fee) { create(:enterprise_fee) }
+    let(:referenced_fee) { create(:enterprise_fee) }
+    let(:product) { create(:product) }
+    let(:distributor) { create(:distributor_enterprise) }
+    let!(:product_distribution) { create(:product_distribution, product: product, distributor: distributor, enterprise_fee: referenced_fee) }
+    let(:current_user) { create(:admin_user) }
+
+    before do
+      allow(controller).to receive(:spree_current_user) { current_user }
+    end
+
+    describe "destroy" do
+      it "removes the fee" do
+        spree_delete :destroy, id: unreferenced_fee.id, format: :json
+        expect(response).to be_success
+        expect { unreferenced_fee.reload }.to raise_error ActiveRecord::RecordNotFound
```

Comment:
RSpec's `change` matcher comes in handy for these cases :point_right: https://relishapp.com/rspec/rspec-expectations/v/3-8/docs/built-in-matchers/change-matcher

Could be implement as

```suggestion
        it "removes the fee" do
          expect { spree_delete :destroy, id: unreferenced_fee.id, format: :json }
            .to change(EnterpriseFee).by(1)
        end
```

I find it a bit more readable as it clearly states what you're actually checking.

Reply:
Does `change(EnterpriseFee).by(1)` work? I always thought it has to be  `change(EnterpriseFee.count).by(1)`.

Reply:
yep, sorry I missed `count` there. You can use either `change { EnterpriseFee.count }` or `change(EntepriseFee, :count)`

Reply:
I did it with change `{ EnterpriseFee.count } by -1` and it works fine !